### PR TITLE
ROX-9331: move access token

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

As an alternative to [pull/10188](https://github.com/stackrox/stackrox/pull/10188) this one moves the short-lived access token to in-memory for the time being.

This has the following caveats:
- For basic auth, a special flow had to be added due to its nature of calling `window.location` doing a hard refresh on the page.
- For any auth provider that is not refresh token enabled (i.e. UserPKI, SAML, Basic); a refresh will trigger a new login, even though the session is still ongoing. These auth providers are the lowest used ones, and basic isn't recommended to be used at all in production.
- As a caveat to the above ^; in case an OIDC provider is used with the fragment mode (e.g. Auth0) where _no_ refresh token is going to be issued, this will have the same effects.

The effects of this is going to be a limited UX for users using SAML, UserPKI, Auth0 auth providers.

Additional research:
The only places _within our application_ that trigger a hard refresh (i.e. assigning `window.location`) is the auth service during basic auth, and in case the telemetry values are changed from enabled to disabled.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

- see CI passing.

Manual testing:

For different types of auth providers, the following actions were manually tested: `Navigate within the UI, Refresh page, open page in new tab`.

| Auth provider                 | Refresh    | New tab    | Naviagate within UI |
|-------------------------------|------------|------------|---------------------|
| Basic | Logged out | Logged out | Logged in           |

| Auth provider                 | Refresh    | New tab    | Naviagate within UI |
|-------------------------------|------------|------------|---------------------|
| Auth0 (OIDC w/ fragment mode) | Logged out | Logged out | Logged in           |

| Auth provider                 | Refresh    | New tab    | Naviagate within UI |
|-------------------------------|------------|------------|---------------------|
| OIDC (w/ post/query mode) | Logged in | Logged in | Logged in           |

| Auth provider                 | Refresh    | New tab    | Naviagate within UI |
|-------------------------------|------------|------------|---------------------|
| OpenShift Auth | Logged in | Logged in | Logged in           |

The following auth providers haven't been tested, but looking through the code should have similar effects:
- SAML & UserPKI - similar to Auth0 / Basic
- IAP - similar to OIDC / OpenShift

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by dhaus67** [comment]
* **#10483** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10483?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dhaus67 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: ui/apps/platform/cypress/integration/login.test.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        interactAndWaitForResponses(() => {
+            cy.get('input[name=username]').type(username);
+            cy.get('input[name=password]').type(password);
+            cy.get('button[type=submit]').click();
+        }, routeMatcherMapForSubmit);
+        cy.location('pathname').should('eq', dashboardURL);
+
+        cy.getAllLocalStorage().then((result) => {
+            expect(result).to.be.empty;
+        });
```
Comment (dvail) [inline ui/apps/platform/cypress/integration/login.test.js:39]: We can see what the test runs show, but I know the main dashboard sets some configuration values in localStorage on load, so there is potential that this check might fail (or fail depending on the results of a race condition).

**Review by dvail — APPROVED** [review]
From a code perspective, if the tests pass this looks good to me.

I'd like to do some manual testing too just to see if this changes anything unexpected in our dev workflows.

**Comment by dhaus67** [comment]
Closing the PR for the time being, since the current approach doesn't work for tokens greater 4KB (which may happen when people make use of the claim mappings functionality).

Will be done at a latter point with introduction of sessions.

**Pull request closed without merging**